File: Easy/Merge Two Sorted Lists/mergetwosortedlists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        pointer1,pointer2 = list1,list2
        current = ListNode(); head = current;
        while pointer1 and pointer2:
            if pointer1.val <= pointer2.val:
                current.next = pointer1
                current = pointer1
                pointer1 = pointer1.next
            else:
                current.next = pointer2
                current = pointer2
                pointer2 = pointer2.next
        if pointer1 or pointer2:
            current.next = pointer1 if pointer1 else pointer2
        return head.next

File: Easy/Merge Two Sorted Lists/mergetwosortedlists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode current = new ListNode();
        ListNode head = current;
        while (list1 != null && list2 != null){
            if (list1.val <= list2.val){
                current.next = list1;
                current = list1;
                list1 = list1.next;
            }
            else{
                current.next = list2;
                current = list2;
                list2 = list2.next;
            }
        }
        if (list1 != null || list2 != null){
            current.next = list1 != null ? list1 : list2;
        }
        return head.next;
    }   
}